drivers/modem/quectel-bg9x.c

```diff
@@ -1188,6 +1196,29 @@ static int modem_init(const struct device *dev)
 			(k_thread_entry_t) modem_rx,
 			NULL, NULL, NULL, K_PRIO_COOP(7), 0, K_NO_WAIT);
 
+
+        /* power on modem */
+        // Enable the battery voltage pin (VLTE-ENB)
+        uint8_t pinPosition = 1 + (8 * 1); // VLTE-ENB is pin 9 on the GPIO expander.
+        uint16_t vLteEnb = 0u;
+        vLteEnb &= ~(1u << pinPosition);
+        vLteEnb |= (1u << pinPosition);
+        uint8_t data[2] = { (uint8_t)(0x00ff & vLteEnb), (uint8_t)((0xff00 & vLteEnb) >> 8) };
+        const struct device *i2cDevice = device_get_binding(IO_EXPANDER_BUS_LABEL);
```

Comment:
Did you get the `i2cDevice`? Because `device_get_binding()` can fail. So check the return pointer before continuing.